core/src/test/java/org/sonarsource/sonarlint/core/container/global/NodeJsHelperTests.java

```diff
@@ -0,0 +1,342 @@
+  void handleErrorDuringVersionCheck() throws IOException {
+    when(globalConfiguration.getNodejsPath()).thenReturn(FAKE_NODE_PATH);
+
+    Queue<ProcessBuilder> pbQueue = new LinkedList<>();
+
+    ProcessBuilder mockedPb1 = mock(ProcessBuilder.class);
+    Process mockedProcess1 = mock(Process.class);
+    when(mockedProcess1.getInputStream()).thenReturn(new ByteArrayInputStream("wrong_version".getBytes(Charset.defaultCharset())));
+    when(mockedPb1.start()).thenReturn(mockedProcess1);
+    pbQueue.add(mockedPb1);
+
+    NodeJsHelper underTest = new NodeJsHelper(globalConfiguration, system2, () -> pbQueue.poll(), null);
+    underTest.start();
+
+    assertThat(logTester.logs()).containsExactly(
+      "Node.js path provided by configuration: " + FAKE_NODE_PATH.toString(),
+      "Checking node version...",
+      "Command '" + FAKE_NODE_PATH.toString() + " -v' returned: wrong_version",
+      "Unable to query node version");
+    assertThat(underTest.getNodeJsPath()).isEqualTo(FAKE_NODE_PATH);
+    assertThat(underTest.getNodeJsVersion()).isNull();
+  }
+
+  @Test
+  void useWhichOnLinuxToResolveNodePath() throws IOException {
+    Queue<ProcessBuilder> pbQueue = new LinkedList<>();
+
+    ProcessBuilder mockedPb1 = mock(ProcessBuilder.class);
+    Process mockedProcess1 = mock(Process.class);
+    when(mockedProcess1.getInputStream()).thenReturn(new ByteArrayInputStream(FAKE_NODE_PATH.toString().getBytes(Charset.defaultCharset())));
+    when(mockedPb1.start()).thenReturn(mockedProcess1);
+    pbQueue.add(mockedPb1);
+
+    ProcessBuilder mockedPb2 = mock(ProcessBuilder.class);
+    Process mockedProcess2 = mock(Process.class);
+    when(mockedProcess2.getInputStream()).thenReturn(new ByteArrayInputStream("v10.5.4".getBytes(Charset.defaultCharset())));
+    when(mockedPb2.start()).thenReturn(mockedProcess2);
+    pbQueue.add(mockedPb2);
+
+    NodeJsHelper underTest = new NodeJsHelper(globalConfiguration, system2, () -> pbQueue.poll(), null);
+    underTest.start();
+
+    assertThat(logTester.logs()).containsExactly(
+      "Looking for node in the PATH",
+      "Command 'which node' returned: " + FAKE_NODE_PATH.toString(),
+      "Found node at " + FAKE_NODE_PATH.toString(),
+      "Checking node version...",
+      "Command '" + FAKE_NODE_PATH.toString() + " -v' returned: v10.5.4");
+    assertThat(underTest.getNodeJsPath()).isEqualTo(FAKE_NODE_PATH);
+    assertThat(underTest.getNodeJsVersion()).isEqualTo(Version.create("10.5.4"));
+  }
+
+  @Test
+  void handleErrorDuringPathCheck() throws IOException {
+    Queue<ProcessBuilder> pbQueue = new LinkedList<>();
+
+    ProcessBuilder mockedPb1 = mock(ProcessBuilder.class);
+    Process mockedProcess1 = mock(Process.class);
+    when(mockedProcess1.getInputStream()).thenReturn(new ByteArrayInputStream("error".getBytes(Charset.defaultCharset())));
+    when(mockedPb1.start()).thenReturn(mockedProcess1);
+    when(mockedProcess1.exitValue()).thenReturn(-1);
+    pbQueue.add(mockedPb1);
```

Comment:
A lot of the same lines are repeated, maybe a method could help initialize all this